packages/shared/lib/clients/provider.client.ts

```diff
@@ -373,6 +388,66 @@ class ProviderClient {
         };
     }
 
+    private async createCorosToken(tokenUrl: string, code: string, clientId: string, clientSecret: string, callback: string): Promise<object> {
+        const headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        };
+        const body = {
+            client_id: clientId,
+            redirect_uri: callback,
+            code: code,
+            client_secret: clientSecret,
+            grant_type: 'authorization_code'
+        };
+        try {
+            const response = await axios.post(tokenUrl, body, { headers: headers });
+            if (response.status === 200 && response.data !== null) {
+                return {
+                    access_token: response.data['access_token'],
+                    refresh_token: response.data['refresh_token'],
+                    expires_in: response.data['expires_in'],
+                    openId: response.data['openId']
+                };
+            }
+
+            throw new NangoError('coros_token_request_error');
+        } catch (e: any) {
+            throw new NangoError('coros_token_request_error', e.message);
+        }
+    }
+
+    private async refreshCorosToken(
+        refreshTokenUrl: string,
+        accessToken: string,
+        refreshToken: string,
+        clientId: string,
+        clientSecret: string
+    ): Promise<object> {
+        const headers = {
+            'Content-Type': 'application/x-www-form-urlencoded'
+        };
+
+        const body = {
+            client_id: clientId,
+            refresh_token: refreshToken,
+            client_secret: clientSecret,
+            grant_type: 'refresh_token'
+        };
+        try {
+            const response = await axios.post(refreshTokenUrl, body, { headers: headers });
+            if (response.status === 200 && response.data !== null && response.data.result === '0000') {
+                return {
+                    access_token: accessToken,
+                    refresh_token: refreshToken,
+                    expires_in: corosExpiresIn
+                };
+            }
+            throw new NangoError('coros_token_refresh_request_error');
```

Comment:
we only have one error for refresh `refresh_token_external_error` + payload (should be json)

Reply:
Thank you for pointing this out 😊!